packages/test/test-end-to-end-tests/src/test/summaries.spec.ts

```diff
@@ -307,6 +324,74 @@ describeNoCompat("Summaries", (getTestObjectProvider) => {
 		);
 	});
 
+	it("should not call entry point before data store is requested", async () => {
+		const dataStoreFactory1 = new DataObjectFactory(
+			TestDataObjectType1,
+			TestDataObject1,
+			[],
+			[],
+		);
+		const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+			runtime.IFluidHandleContext.resolveHandle(request);
+		const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+			[dataStoreFactory1.type, Promise.resolve(dataStoreFactory1)],
+		]);
+		const containerRuntimeFactoryWithDefaultDataStore =
+			getContainerRuntimeApi(pkgVersion).ContainerRuntimeFactoryWithDefaultDataStore;
+		const runtimeFactory = new containerRuntimeFactoryWithDefaultDataStore(
+			dataStoreFactory1,
+			registryStoreEntries,
+			undefined,
+			[innerRequestHandler],
+		);
+		const loader = provider.createLoader([[provider.defaultCodeDetails, runtimeFactory]]);
```

Comment:
You don't need to create a loader. You can directly call `provider.createContainer(runtimeFactory)`. Take a look at [these tests](https://github.com/agarwal-navin/FluidFramework/blob/6af5a9369ca7f0a52e82e53a52d99246b649a11f/packages/test/test-end-to-end-tests/src/test/gc/gcReferenceUpdatesInSummary.spec.ts#L90) and how it creates a container. You can do the same minus the runtimeOptions which are not needed.

Reply:
Modified